drivers/mfd/tswizard-core.c

```diff
@@ -291,44 +291,45 @@ static int ts_supervisor_i2c_probe(struct i2c_client *client)
 			dev_warn(dev, "error creating sysfs entries\n");
 	}
 
-	if (features & SUPER_FEAT_SN) {
+	if (features & WIZARD_FEAT_SN) {
 		err = sysfs_create_group(&dev->kobj, &serial_attr_group);
 		if (err)
 			dev_warn(dev, "error creating sysfs entries\n");
 	}
 
 	/* Set up and register the platform devices. */
-	for (i = 0; i < ARRAY_SIZE(tssupervisor_devs); i++) {
-		tssupervisor_devs[i].platform_data = super;
-		tssupervisor_devs[i].pdata_size = sizeof(struct ts_supervisor);
+	for (i = 0; i < ARRAY_SIZE(tswizard_devs); i++) {
+		tswizard_devs[i].platform_data = wizard;
+		tswizard_devs[i].pdata_size = sizeof(struct ts_wizard);
 	}
 
-	return mfd_add_devices(dev, 0, tssupervisor_devs,
-			      ARRAY_SIZE(tssupervisor_devs), NULL, 0, NULL);
+	return mfd_add_devices(dev, 0, tswizard_devs,
+			      ARRAY_SIZE(tswizard_devs), NULL, 0, NULL);
 }
 
-static const struct i2c_device_id ts_supervisor_i2c_id[] = {
-	{ "tssupervisor", 0 },
+static const struct i2c_device_id ts_wizard_i2c_id[] = {
+	{ "tswizard", 0 },
 	{ /* sentinel */ }
 };
-MODULE_DEVICE_TABLE(i2c, ts_supervisor_i2c_id);
+MODULE_DEVICE_TABLE(i2c, ts_wizard_i2c_id);
 
-static const struct of_device_id ts_supervisor_i2c_of_match[] = {
-	{ .compatible = "technologic,supervisor", },
+static const struct of_device_id ts_wizard_i2c_of_match[] = {
+	{ .compatible = "technologic,tsupervisor", },
+	{ .compatible = "technologic,tswizard" },
```

Comment:
Leaving old compatible in is a great idea, I didn't notice any other drivers doing that. Is that something we should be doing across all drivers? And also, this breaks the old compatible string anyway, so, this may not be as intended?

Reply:
There ended up not being a good way to do that as we looked at it more. The MFD driver has to advertise what the sub devices are in C code, and list the matching compatible strings.  There is only 1 compatible string you can list, so we can't support both without maintaining two entirely separate lists of mfd devices.  IMO we should just suggest the change at that point.

I'd also suggest we just remove tssupervisor from here even if we aren't going to support that.